frontend/src/metabase/components/FieldValuesWidget.jsx

```diff
@@ -0,0 +1,323 @@
+/* @flow */
+
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { t } from "c-3po";
+
+import TokenField from "metabase/components/TokenField";
+import RemappedValue from "metabase/containers/RemappedValue";
+import LoadingSpinner from "metabase/components/LoadingSpinner";
+
+import { MetabaseApi } from "metabase/services";
+import { addRemappings, fetchFieldValues } from "metabase/redux/metadata";
+import { defer } from "metabase/lib/promise";
+import { debounce } from "underscore";
+import { stripId } from "metabase/lib/formatting";
+
+import type Field from "metabase-lib/lib/metadata/Field";
+import type { FieldId } from "metabase/meta/types/Field";
+import type { Value } from "metabase/meta/types/Dataset";
+
+const MAX_SEARCH_RESULTS = 100;
+
+const mapDispatchToProps = {
+  addRemappings,
+  fetchFieldValues,
+};
+
+type Props = {
+  value: Value[],
+  onChange: (value: Value[]) => void,
+  field: Field,
+  searchField?: Field,
+  multi?: boolean,
+  autoFocus?: boolean,
+  color?: string,
+  fetchFieldValues: (id: FieldId) => void,
+  maxResults: number,
+  style?: { [key: string]: string | number },
```

Comment:
What's the main reason for allowing style as props here? If we can find a way to control specific instance needs without this I'd prefer that rather than allowing arbitrary style.